### PR TITLE
Users/akata72/scalingplans

### PR DESCRIPTION
**Pull request by akata72**
# Change

Azure Virtual Desktop ScalingPlans. (Links to #1131)
image

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

**Comment by akata72** [comment]
This is the scalingplans introduced for the Azure Virtual Desktop service. Please take a look and see if it can be used. Thanks!

File: README.md [hunk 1]
```diff
@@ -28,97 +28,98 @@ The platform supports both ARM and Bicep and can be leveraged using GitHub actio
...
-| [Web StaticSites](https://github.com/Azure/ResourceModules/tree/main/arm/Microsoft.Web/staticSites) | [!['Web: StaticSites'](https://github.com/Azure/ResourceModules/actions/workflows/ms.web.staticsites.yml/badge.svg)](https://github.com/Azure/ResourceModules/actions/workflows/ms.web.staticsites.yml) |
-| [Web/Function Apps](https://github.com/Azure/ResourceModules/tree/main/arm/Microsoft.Web/sites) | [!['Web: Sites'](https://github.com/Azure/ResourceModules/actions/workflows/ms.web.sites.yml/badge.svg)](https://github.com/Azure/ResourceModules/actions/workflows/ms.web.sites.yml) |
+| [Action Groups](https://github.com/akata72/ResourceModules/tree/main/arm/Microsoft.Insights/actionGroups) | [!['Insights: ActionGroups'](https://github.com/akata72/ResourceModules/actions/workflows/ms.insights.actiongroups.yml/badge.svg)](https://github.com/akata72/ResourceModules/actions/workflows/ms.insights.actiongroups.yml) |
```
Comment (MrMCake) [inline README.md:31]: Please undo by copy the content [here](https://raw.githubusercontent.com/Azure/ResourceModules/main/README.md) into your file. The ReadMe is updated automatically which has its ups - and downs ;)

File: .github/workflows/ms.desktopvirtualization.scalingplans.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
+name: 'DesktopVirtualization: Scalingplans'
```
Comment (MrMCake) [inline .github/workflows/ms.desktopvirtualization.scalingplans.yml:1]: Please also create a copy in the `.azuredevops/modulePipelines` directory. Just as the GitHub pipelines you can create a simple copy of one of the other pipelines and update the values accordingly

File: arm/Microsoft.DesktopVirtualization/scalingplans/readme.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
+# DesktopVirtualization ScalingPlans `[Microsoft.DesktopVirtualization/scalingPlans]`
+
+This module deploys DesktopVirtualization ScalingPlans.
+// TODO: Replace Resource and fill in description
```
Comment (MrMCake) [inline arm/Microsoft.DesktopVirtualization/scalingplans/readme.md:4]: Remove placeholder

File: arm/Microsoft.DesktopVirtualization/scalingplans/readme.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+| `name` | string |  |  | Required. Name of the scaling plan |
+| `roleAssignments` | array | `[]` |  | Optional. Array of role assignment objects that contain the 'roleDefinitionIdOrName' and 'principalIds' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: '/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11' |
+| `scalingplanDescription` | string | `[parameters('name')]` |  | Optional. Description of the scaling plan. |
+| `schedules` | array | `[System.Collections.Hashtable]` |  | Optional. The schedules related to this scaling plan. If no value is provided a default schedule will be provided. |
+| `tags` | object | `{object}` |  | Optional. Tags of the resource. |
+| `timeZone` | string | `W. Europe Standard Time` |  | Optional. Timezone to be used for the scaling plan. |
+
+### Parameter Usage: `<ParameterPlaceholder>`
+
+// TODO: Fill in Parameter usage
```
Comment (MrMCake) [inline arm/Microsoft.DesktopVirtualization/scalingplans/readme.md:38]: Please remove placeholder

File: arm/README.md [hunk 5]
```diff
@@ -4,94 +4,95 @@ In this section you can find useful information regarding the Modules that are c
...
-| [App Service Environments](https://github.com/Azure/ResourceModules/tree/main/arm/Microsoft.Web/hostingEnvironments) |  | [hostingEnvironments](https://github.com/Azure/ResourceModules/tree/main/arm/Microsoft.Web/hostingEnvironments) |
-| [App Service Plans](https://github.com/Azure/ResourceModules/tree/main/arm/Microsoft.Web/serverfarms) |  | [serverfarms](https://github.com/Azure/ResourceModules/tree/main/arm/Microsoft.Web/serverfarms) |
-| [Web/Function Apps](https://github.com/Azure/ResourceModules/tree/main/arm/Microsoft.Web/sites) |  | [sites](https://github.com/Azure/ResourceModules/tree/main/arm/Microsoft.Web/sites) |
-| [Web StaticSites](https://github.com/Azure/ResourceModules/tree/main/arm/Microsoft.Web/staticSites) |  | [staticSites](https://github.com/Azure/ResourceModules/tree/main/arm/Microsoft.Web/staticSites) |
+| [Analysis Services Server](https://github.com/akata72/ResourceModules/tree/main/arm/Microsoft.AnalysisServices/servers) | `MS.AnalysisServices` | [servers](https://github.com/akata72/ResourceModules/tree/main/arm/Microsoft.AnalysisServices/servers) |
```
Comment (MrMCake) [inline arm/README.md:7]: Same comment as above. Please replace with content [this](https://raw.githubusercontent.com/Azure/ResourceModules/main/arm/README.md) file.

**Comment by MrMCake** [comment]
Hey @akata72, thanks for the awesome contribution, just a few comments. As its a new module there are always some initial cleanup tasks that won't occur when update an existing module.

**Review by MariusStorhaug — CHANGES_REQUESTED** [review]

File: arm/Microsoft.DesktopVirtualization/scalingplans/readme.md (hunk 4 above)
Comment (MariusStorhaug) [reply arm/Microsoft.DesktopVirtualization/scalingplans/readme.md:38]: Well, either remove or add details of the array/object parameters.

**Review by MariusStorhaug — APPROVED** [review]

**Pull request merged**